src/orders/components/OrderDetailsRefundTable/refunds.ts

```diff
@@ -49,21 +61,46 @@ export const manualRefundsExtractor = (
     event => !idsOfEventsAssociatedToGrantedRefunds.has(event.id),
   );
 
-  // Map filtered events to DatagridRefund objects
-  return manualRefundEvents?.map(event => ({
-    id: event.id,
-    type: "manual",
-    status:
-      event.type === "REFUND_SUCCESS"
-        ? OrderGrantedRefundStatusEnum.SUCCESS
-        : OrderGrantedRefundStatusEnum.FAILURE,
-    amount: event.amount,
-    createdAt: event.createdAt,
-    user: {
-      email: determineCreatorDisplay(event.createdBy),
-    },
-    reason: intl.formatMessage(refundGridMessages.manualRefund),
-  }));
+  // Group events by pspReference
+  const eventsByPspReference = manualRefundEvents?.reduce<
+    Record<string, TransactionEventFragment[]>
+  >((acc, event) => {
+    if (!acc[event.pspReference]) {
+      acc[event.pspReference] = [];
+    }
+
+    acc[event.pspReference].push(event);
+
+    return acc;
+  }, {});
+
+  // Map grouped events to DatagridRefund objects, taking the latest status
+  const datagridRefunds = Object.values(eventsByPspReference).map(eventGroup => {
+    const sortedEvents = eventGroup.sort(
+      (a, b) => new Date(b.createdAt).getTime() - new Date(a.createdAt).getTime(),
+    );
+    const latestEvent = sortedEvents[0];
+    const latestEventWithAuthor = findLatestEventWithCreatedBy(sortedEvents) || latestEvent;
+
+    return {
+      id: latestEvent.id,
+      type: "manual" as const,
+      status:
+        latestEvent.type === "REFUND_SUCCESS"
+          ? OrderGrantedRefundStatusEnum.SUCCESS
+          : latestEvent.type === "REFUND_FAILURE"
+            ? OrderGrantedRefundStatusEnum.FAILURE
+            : OrderGrantedRefundStatusEnum.PENDING,
```

Comment:
issue: create function that obtain status, this is really hard to read, considering multi-nested in-line if